test/tools_tests/branch_length_estimator_tests/iid_exponential_mle_test.py

```diff
@@ -359,3 +360,244 @@ def test_minimum_branch_length(self, name, solver):
+                    tree.get_branch_length("0", "1"), branch1, places=3
+                )
+            with self.assertRaises(AssertionError):
+                self.assertAlmostEqual(
+                    tree.get_branch_length("1", "2"), branch2, places=3
+                )
+            self.assertAlmostEqual(tree.get_time("0"), 0.0, places=3)
+            with self.assertRaises(AssertionError):
+                self.assertAlmostEqual(tree.get_time("1"), branch1, places=3)
+            self.assertAlmostEqual(tree.get_time("2"), 1.0, places=3)
+            for x, y in zip(model.mutation_rate, mutation_rates):
+                with self.assertRaises(AssertionError):
+                    self.assertAlmostEqual(x, y, places=3)
+
+    def test_invalid_site_rates(self):
+        """
+        Tree topology is the same as test_hand_solvable_problem_with_site_rate
+        but rates are misspecified so we should error out.
+        """
+        rate_1, rate_2, rate_3 = 1.5, -1, 2.5
+
+        tree = nx.DiGraph()
+        tree.add_nodes_from(["0", "1", "2"])
+        tree.add_edge("0", "1")
+        tree.add_edge("1", "2")
+        tree = CassiopeiaTree(tree=tree)
+        tree.set_all_character_states(
+            {"0": [0, 0, 0], "1": [1, 0, 0], "2": [1, 1, 0]}
+        )
+        relative_rates = [rate_1, rate_2, rate_3]
+        model = IIDExponentialMLE(
+            minimum_branch_length=1e-4,
+            relative_mutation_rates=relative_rates,
+        )
+        with self.assertRaises(ValueError):
+            model.estimate_branch_lengths(tree)
+
+    @parameterized.expand(
+        [
+            (
+                "should_pass",
+                [1.5, 2, 2.5, 1.5, 2, 2.5],
+                [1.5, 2, 2.5, 1.5, 2, 2.5],
+            ),
+            (
+                "should_not_pass",
+                [1.5, 2, 2.5, 1.5, 2, 2.5],
+                [1.52, 1.98, 2.48, 1.52, 2.01, 2.49],
+            ),
+        ]
+    )
+    def test_larger_hand_solvable_problem_with_site_rates(
+        self, name, solver_rates, math_rates
+    ):
+        """
+        Tree topology is a duplicated version of
+        test_hand_solvable_problem_with_site_rates. That is, we double the
+        number of characters (while using the same site rates for each pair)
+        and decouple each using missing characters as shown below. The expected
+        result is the same as the aforementioned test.
+
+        The structure:
+                   root [state = '0000000']
+                    |
+                    x
+                  child [state = '100100']
+                    |
+           |------------------|
+           y                  z
+        [state=            [state=
+         XXX110]            110XXX]
+        """
+        rate_1, rate_2, rate_3, rate_4, rate_5, rate_6 = solver_rates
+        (
+            math_rate_1,
+            math_rate_2,
+            math_rate_3,
+            math_rate_4,
+            math_rate_5,
+            math_rate_6,
```

Comment:
Since `math_rate_4, math_rate_5, math_rate_6` are never used, you can just change them by `_`, which is the convention for unused variables.